libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxManeuverApi.kt

```diff
@@ -0,0 +1,100 @@
+package com.mapbox.navigation.ui.maneuver.api
+
+import com.mapbox.api.directions.v5.models.BannerComponents
+import com.mapbox.api.directions.v5.models.BannerInstructions
+import com.mapbox.api.directions.v5.models.LegStep
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.navigation.base.formatter.DistanceFormatter
+import com.mapbox.navigation.base.trip.model.RouteStepProgress
+import com.mapbox.navigation.ui.base.api.maneuver.ManeuverApi
+import com.mapbox.navigation.ui.base.api.maneuver.ManeuverCallback
+import com.mapbox.navigation.ui.base.api.maneuver.StepDistanceRemainingCallback
+import com.mapbox.navigation.ui.base.api.maneuver.UpcomingManeuversCallback
+import com.mapbox.navigation.ui.base.model.maneuver.Maneuver
+import com.mapbox.navigation.ui.base.model.maneuver.ManeuverState
+import com.mapbox.navigation.ui.base.model.maneuver.ManeuverState.CurrentManeuver
+import com.mapbox.navigation.ui.base.model.maneuver.ManeuverState.DistanceRemainingToFinishStep
+import com.mapbox.navigation.ui.maneuver.ManeuverAction
+import com.mapbox.navigation.ui.maneuver.ManeuverProcessor
+import com.mapbox.navigation.ui.maneuver.ManeuverResult
+import com.mapbox.navigation.utils.internal.JobControl
+import com.mapbox.navigation.utils.internal.ThreadController
+import kotlinx.coroutines.launch
+
+/**
+ * Implementation of [ManeuverApi] allowing access to generate representation of
+ * maneuvers based on [BannerInstructions]
+ * @property distanceFormatter DistanceFormatter
+ */
+class MapboxManeuverApi internal constructor(
```

Comment:
This makes sense, but I am not sure what are we exactly canceling, because none of the API calls are only responsible to request road shields. Each API is doing much more than just requesting road shields. So what happens exactly when someone calls cancel?

Reply:
Added cancel calls.